packages/sanity/src/core/network/isUsingModernHttp.ts

```diff
@@ -0,0 +1,128 @@
+import {type SanityClient} from '@sanity/client'
+import {from, Observable, of, race, timer} from 'rxjs'
+import {map, take} from 'rxjs/operators'
+
+/**
+ * The /ping route is special in that it allows any origin to access it, and crucially
+ * for this case: also allows _timing_ requests through the `timing-allow-origin` header.
+ * This is what allows us to detect the protocol used for the request by using the
+ * browser Performance API.
+ */
+const checkPath = '/ping'
+
+/**
+ * The tag to use for the request to the /ping route. This is used to identify the
+ * request in the Performance API as well as in the server logs.
+ */
+const checkRequestTag = 'protocol-check'
+
+/**
+ * Preferably we want to use a client, since that will have both a configured URL (the
+ * user may have configured a custom API URL) and also uses the client for _requesting_
+ * the URL. While this last part _shouldn't_ make a difference, there is a theoretical
+ * possibility that XMLHttpRequest (potentially used by the client) uses a different
+ * protocol than the fetch API.
+ */
+const fallbackApiUrl = `https://api.sanity.io/v2025-03-01/ping?tag=sanity.studio.${checkRequestTag}`
```

Comment:
I guess in theory a VPN/proxy could allow http/2 on `api.sanity.io` but still degrade to http/1 on `<projectId>.api.sanity.io` or vice versa?

Does this request also risk getting blocked by in-flight requests that now are stalled because of degraded protocol?